src/web_server.h

```diff
@@ -1,36 +1,78 @@
 #ifndef NETDATA_WEB_SERVER_H
 #define NETDATA_WEB_SERVER_H 1
 
-#define WEB_PATH_FILE               "file"
-#define WEB_PATH_DATA               "data"
-#define WEB_PATH_DATASOURCE         "datasource"
-#define WEB_PATH_GRAPH              "graph"
+/**
+ * @file web_server.h
+ * @brief API of the web server.
+ */
 
-#define LISTEN_PORT 19999
-#define LISTEN_BACKLOG 100
+#define WEB_PATH_FILE               "file"       ///< web path file string
+#define WEB_PATH_DATA               "data"       ///< web path data string
+#define WEB_PATH_DATASOURCE         "datasource" ///< web path datasource string
+#define WEB_PATH_GRAPH              "graph"      ///< web path graph string 
+
+#define LISTEN_PORT 19999  ///< default port to listen
+#define LISTEN_BACKLOG 100 ///< default backlog to listen
 
 #ifndef MAX_LISTEN_FDS
-#define MAX_LISTEN_FDS 100
+#define MAX_LISTEN_FDS 100 ///< default maximum file descriptors to listen to
 #endif
 
-#define WEB_SERVER_MODE_MULTI_THREADED 0
-#define WEB_SERVER_MODE_SINGLE_THREADED 1
+#define WEB_SERVER_MODE_MULTI_THREADED 0  ///< Multi threaded mode.
+#define WEB_SERVER_MODE_SINGLE_THREADED 1 ///< Single threaded mode.
+/// WEB_SERVER_MODE_*
 extern int web_server_mode;
 
+/**
+ * Main method of thread of the web server.
+ *
+ * This starts a multi threaded web server.
+ *
+ * @param ptr to struct netdata_static_thread
+ * @return NULL
+ */
 extern void *socket_listen_main_multi_threaded(void *ptr);
+/**
+ * Main method of thread of the web server.
+ *
+ * This starts a single threaded web server.
+ *
+ * @param ptr to struct netdata_static_thread
+ * @return NULL
+ */
 extern void *socket_listen_main_single_threaded(void *ptr);
+/**
+ * Create sockets to listen to.
+ *
+ * @return the number of open sockets.
+ */
 extern int create_listen_sockets(void);
+/**
+ * Check if `fd` is in the list sockets listening for web server requests.
+ *
+ * @param fd to check
+ * @return boolean
+ */
 extern int is_listen_socket(int fd);
 
 #ifndef HAVE_ACCEPT4
+/**
+ * ktsaou: Your help needed.
```

Comment:
this is a replacement for `accept4()` for systems that do not have one. `man accept4`.